octodns/provider/yaml.py

```diff
@@ -235,7 +254,8 @@ def _apply(self, plan):
                 del d['ttl']
             if record._octodns:
                 d['octodns'] = record._octodns
-            data[record.name].append(d)
+            # we want to output the utf-8 version of the name
+            data[record.decoded_name].append(d)
```

Comment:
Going forward all record names are written out in utf-8, decoded from idna if needed.